lib/services/dialog_import_service.rb

```diff
@@ -99,12 +99,14 @@ def check_field_associations(fields)
   end
 
   def import(dialog)
-    @dialog_import_validator.determine_dialog_validity(dialog)
-    new_dialog = Dialog.create(dialog.except('dialog_tabs', 'export_version'))
-    association_list = build_association_list(dialog)
-    new_dialog.update!(dialog.merge('dialog_tabs' => build_dialog_tabs(dialog, dialog['export_version'] || DEFAULT_DIALOG_VERSION)))
-    build_associations(new_dialog, association_list)
-    new_dialog
+    ActiveRecord::Base.transaction do
+      @dialog_import_validator.determine_dialog_validity(dialog)
+      new_dialog = Dialog.create(dialog.except('dialog_tabs', 'export_version'))
+      association_list = build_association_list(dialog)
+      new_dialog.update!(dialog.merge('dialog_tabs' => build_dialog_tabs(dialog, dialog['export_version'] || DEFAULT_DIALOG_VERSION)))
+      build_associations(new_dialog, association_list)
```

Comment:
Looks like `build_associations` does `DialogFieldAssociation.create` which would also silently fail on error, causing the dialog to be created with missing associations.

Can you see what's being called in this whole transaction block and ensure  `create` becomes `create!` and the same for `update` `save`, etc?

Can you add to your test where DialogFieldAssociation.create fails validation to make sure they're currently "broken" and then fixed with any of your changes?